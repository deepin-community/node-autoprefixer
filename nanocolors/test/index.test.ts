import * as main from '../index'

it('prints colors', () => {
  let colors = main.createColors(true)
  expect(colors.isColorSupported).toBe(true)
  let output = Object.entries(colors)
    .filter(([name]) => name !== 'isColorSupported' && name !== 'reset')
    .reduce(
      (all, [name, fn]) => `${all}\n${name.padEnd(13)} ${fn(name)}`,
      `\x1b[1mreset         ${colors.reset('reset')}`
    )
  expect(output).toMatchSnapshot()
})

it('supports nested colors', () => {
  let { green, red, bold, dim } = main.createColors(true)
  expect(green('a ' + red('b') + ' c')).toMatchSnapshot()
  expect(bold('b ' + red('r ' + dim('d') + ' r') + ' b')).toMatchSnapshot()
})

it('disables colors', () => {
  let noColors = main.createColors(false)
  expect(noColors.isColorSupported).toBe(false)
  let keys = Object.keys(noColors) as (keyof main.Colors)[]
  keys.forEach(name => {
    if (name !== 'isColorSupported') {
      expect(noColors[name]('text')).toEqual('text')
    }
  })
})

it('copies all methods', () => {
  let created = Object.keys(main.createColors(true))
  expect(Object.keys(main).sort()).toEqual(
    created.concat(['createColors']).sort()
  )
})

it('does not add modifiers to empty string', () => {
  let { green } = main.createColors(true)
  expect(green('')).toEqual('')
})

it('converts non-string to string', () => {
  let { red } = main.createColors(true)
  expect(typeof red(1)).toEqual('string')
  expect(red(1)).toEqual(red('1'))
  // @ts-expect-error
  expect(red(new Date(0))).toEqual(red(String(new Date(0))))
  // @ts-expect-error
  expect(red()).toEqual(red('undefined'))
  // @ts-expect-error
  expect(red(undefined)).toEqual(red('undefined'))
  // @ts-expect-error
  expect(red(null)).toEqual(red('null'))
})

it('converts non-string to string on disabled colors', () => {
  let { red } = main.createColors(false)
  expect(typeof red(1)).toEqual('string')
  expect(red(1)).toEqual('1')
  // @ts-expect-error
  expect(red()).toEqual('undefined')
  // @ts-expect-error
  expect(red()).toEqual('undefined')
  // @ts-expect-error
  expect(red(null)).toEqual('null')
})
