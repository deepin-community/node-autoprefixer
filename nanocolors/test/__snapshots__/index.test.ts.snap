// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prints colors 1`] = `
"[1mreset         [0mreset[0m
bold          [1mbold[22m
dim           [2mdim[22m
italic        [3mitalic[23m
underline     [4munderline[24m
inverse       [7minverse[27m
hidden        [8mhidden[28m
strikethrough [9mstrikethrough[29m
black         [30mblack[39m
red           [31mred[39m
green         [32mgreen[39m
yellow        [33myellow[39m
blue          [34mblue[39m
magenta       [35mmagenta[39m
cyan          [36mcyan[39m
white         [37mwhite[39m
gray          [90mgray[39m
bgBlack       [40mbgBlack[49m
bgRed         [41mbgRed[49m
bgGreen       [42mbgGreen[49m
bgYellow      [43mbgYellow[49m
bgBlue        [44mbgBlue[49m
bgMagenta     [45mbgMagenta[49m
bgCyan        [46mbgCyan[49m
bgWhite       [47mbgWhite[49m"
`;

exports[`supports nested colors 1`] = `"[32ma [31mb[32m c[39m"`;

exports[`supports nested colors 2`] = `"[1mb [31mr [2md[22m[1m r[39m b[22m"`;
